tui/src/ui/components/database.rs

```diff
@@ -618,46 +619,64 @@ impl Model {
         self.database_sync_results();
     }
 
-    pub fn database_update_search(&mut self, input: &str) {
+    pub fn update_search<T: Indexable>(&self, indexable_songs: &Vec<T>, input: &str) -> Table {
         let mut table: TableBuilder = TableBuilder::default();
-        let mut idx = 0;
         let search = format!("*{}*", input.to_lowercase());
-        let mut db_tracks = vec![];
-        if let Ok(tracks) = self.db.get_all_records() {
-            db_tracks.clone_from(&tracks);
-            for record in tracks {
-                if wildmatch::WildMatch::new(&search).matches(&record.artist.to_lowercase())
-                    | wildmatch::WildMatch::new(&search).matches(&record.title.to_lowercase())
-                {
-                    if idx > 0 {
-                        table.add_row();
-                    }
-
-                    let duration =
-                        termusiclib::track::Track::duration_formatted_short(&record.duration);
-                    let duration_string = format!("[{duration:^6.6}]");
-
-                    table
-                        .add_col(TextSpan::new(duration_string.as_str()))
-                        .add_col(
-                            TextSpan::new(record.artist)
-                                .fg(tuirealm::ratatui::style::Color::LightYellow),
-                        )
-                        .add_col(TextSpan::new(record.title).bold())
-                        .add_col(TextSpan::new(record.file));
-                    // .add_col(TextSpan::new(record.album().unwrap_or("Unknown Album")));
-                    idx += 1;
+        let mut idx = 0;
+        for record in indexable_songs {
+            let artist_match: bool = if let Some(artist) = record.meta_artist() {
+                wildmatch::WildMatch::new(&search).matches(&artist.to_lowercase())
+            } else {
+                false
+            };
+            let title_match: bool = if let Some(title) = record.meta_title() {
+                wildmatch::WildMatch::new(&search).matches(&title.to_lowercase())
+            } else {
+                false
+            };
+            let album_match: bool = if let Some(album) = record.meta_album() {
+                wildmatch::WildMatch::new(&search).matches(&album.to_lowercase())
+            } else {
+                false
+            };
+            if artist_match | title_match | album_match {
```

Comment:
```suggestion
            if artist_match || title_match || album_match {
```

Make it a logical OR instead of a binary OR

Reply:
done